tests/components/home_connect/conftest.py

```diff
@@ -0,0 +1,316 @@
+async def mock_integration_setup(
+    hass: HomeAssistant,
+    platforms: list[Platform],
+    config_entry: MockConfigEntry,
+) -> Callable[[], Awaitable[bool]]:
+    """Fixture to set up the integration."""
+    config_entry.add_to_hass(hass)
+
+    async def run() -> bool:
+        with patch("homeassistant.components.home_connect.PLATFORMS", platforms):
+            result = await hass.config_entries.async_setup(config_entry.entry_id)
+            await hass.async_block_till_done()
+        return result
+
+    return run
+
+
+@pytest.fixture(name="get_appliances")
+def mock_get_appliances() -> Generator[None, Any, None]:
+    """Mock ConfigEntryAuth parent (HomeAssistantAPI) method."""
+    with patch(
+        "homeassistant.components.home_connect.api.ConfigEntryAuth.get_appliances",
+    ) as mock:
+        yield mock
+
+
+@pytest.fixture(name="appliance")
+def mock_appliance(request) -> Mock:
+    """Fixture to mock Appliance."""
+    app = "Washer"
+    if hasattr(request, "param") and request.param:
+        app = request.param
+
+    mock = MagicMock(
+        autospec=HomeConnectAppliance,
+        **MOCK_APPLIANCES_PROPERTIES.get(app),
+    )
+    mock.name = app
+    type(mock).status = PropertyMock(return_value={})
+    mock.get.return_value = {}
+    mock.get_programs_available.return_value = []
+    mock.get_status.return_value = {}
+    mock.get_settings.return_value = {}
+
+    return mock
+
+
+@pytest.fixture(name="problematic_appliance")
+def mock_problematic_appliance(
+    appliance_name: str = "Washer",
+) -> Generator[Mock, None, None]:
+    """Fixture to mock a problematic Appliance."""
+    mock = Mock(
+        spec=HomeConnectAppliance,
+        **MOCK_APPLIANCES_PROPERTIES.get(appliance_name),
+    )
+    mock.name = appliance_name
+    setattr(mock, "status", {})
+    mock.get_programs_active.side_effect = HomeConnectError
+    mock.get_programs_available.side_effect = HomeConnectError
+    mock.start_program.side_effect = HomeConnectError
+    mock.stop_program.side_effect = HomeConnectError
+    mock.get_status.side_effect = HomeConnectError
+    mock.get_settings.side_effect = HomeConnectError
+    mock.set_setting.side_effect = HomeConnectError
+
+    with patch(
+        "homeconnect.api.HomeConnectAppliance",
+        return_value=mock,
+    ):
+        yield mock
+
+
+@pytest.fixture(name="config_entry_auth")
+def mock_config_entry_auth():
+    """Fixture to mock the ConfigEntryAuth class."""
+    with patch(
+        "homeassistant.components.home_connect.api.ConfigEntryAuth",
+        autospec=MockConfigEntryAuth,
+    ):
+        yield
+
+
+@pytest.fixture(name="config_entry_auth_devices")
+def mock_config_entry_auth_devices():
+    """Fixture to mock ConfigEntryAuth class with another class."""
+    with patch(
+        "homeassistant.components.home_connect.api.ConfigEntryAuth",
+        new_callable=MockConfigEntryAuth,
+    ):
+        yield
+
+
+def get_appliances():
```

Comment:
Please clarify why we have this when we also have a fixture for `appliance` eg. Is this function needed when we want to mock more than a single appliance? Maybe change the name to make that more clear?